lib/android_safetynet/attestation_response.rb

```diff
@@ -45,6 +45,10 @@ def certificate_chain
       end
     end
 
+    def timestamp
+      Time.at((payload["timestampMs"] / 1000.0).round)
```

Comment:
So `AndroidSafetynet::AttestationResponse` only performs the parsing of the timestamp so far. It's not considering itself invalid or valid based on the timestamp. Was that intentional or should we make the timestamp validation part of the `#verify` above?

Reply:
It was intentional, though I did have some doubts. I tried to make it work the way `AndroidSafetynet::AttestationResponse#certificate_chain` works with `WebAuthn::AttestationStatement::AndroidSafetynet#trusted_attestation_certificate?` but now that I take another look at it I see that the latter should be moved.

I guess that even if `trusted_attestation_certificate?` becomes a responsibility of `AndroidSafetynet::AttestationResponse` the `certificate_chain` would still be used to return the `attestation_certificate` so it's different from `timestamp`. I'll follow your suggestion and make the timestamp validation part of the `#verify`.

Still not sure why the check on the ctsProfileMatch is not done on the `#verify` as well 🤔

Reply:
> It was intentional, though I did have some doubts. I tried to make it work the way `AndroidSafetynet::AttestationResponse#certificate_chain` works with `WebAuthn::AttestationStatement::AndroidSafetynet#trusted_attestation_certificate?` but now that I take another look at it I see that the latter should be moved.
> 
> I guess that even if `trusted_attestation_certificate?` becomes a responsibility of `AndroidSafetynet::AttestationResponse` the `certificate_chain` would still be used to return the `attestation_certificate` so it's different from `timestamp`. I'll follow your suggestion and make the timestamp validation part of the `#verify`.

Gotcha.
Sounds good.

> 
> Still not sure why the check on the ctsProfileMatch is not done on the `#verify` as well thinking

Yes, that's something to further think about.

Initially it was left out because it sounded like an extra integrity check in the SafetyNet docs, and the WebAuthn spec explicitly listed it as an extra check needed for WebAuthn (https://www.w3.org/TR/webauthn/#android-safetynet-attestation).